include/core/CJsonStateRestoreTraverser.h

```diff
@@ -118,19 +118,16 @@ class CORE_EXPORT CJsonStateRestoreTraverser : public CStateRestoreTraverser {
         SBoostJsonHandler();
 
         //! The maximum number of elements allowed in an array
-        static constexpr std::size_t max_array_size =
-            std::numeric_limits<std::size_t>::max();
+        static constexpr std::size_t max_array_size = 1000000; // 1 million elements
```

Comment:
Nit: you can write ` max_array_size = 1'000'000` to make the number more readable, and then drop the comment.